website/src/components/Group/Inputs.tsx

```diff
@@ -0,0 +1,236 @@
+    <TextInput
+        className='sm:col-span-4'
+        type='email'
+        label='Contact email address'
+        name='email'
+        fieldMappingKey='contactEmail'
+        defaultValue={defaultValue}
+    />
+);
+
+type CountryInputProps = {
+    defaultValue?: string;
+};
+
+export const CountryInput: FC<CountryInputProps> = ({ defaultValue }) => (
+    <LabelledInputContainer label='Country' htmlFor='country' className='sm:col-span-3' required>
+        <select
+            id={fieldMapping.country.id}
+            name='country'
+            required={fieldMapping.country.required}
+            autoComplete='country-name'
+            className='block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 focus:ring-2 focus:ring-inset focus:ring-primary-600 sm:max-w-xs sm:text-sm sm:leading-6'
+            defaultValue={defaultValue}
+        >
+            <option>{chooseCountry}</option>
+            {listOfCountries.map((country) => (
+                <option key={country} value={country}>
+                    {country}
+                </option>
+            ))}
+        </select>
+    </LabelledInputContainer>
+);
+
+type AddressLineOneInputProps = {
+    defaultValue?: string;
+};
+
+export const AddressLineOneInput: FC<AddressLineOneInputProps> = ({ defaultValue }) => (
+    <TextInput
+        className='col-span-full'
+        type='text'
+        label='Address Line 1'
+        name='address-line-1'
+        fieldMappingKey='line1'
+        defaultValue={defaultValue}
+    />
+);
+
+type AddressLineTwoInputProps = {
+    defaultValue?: string;
+};
+
+export const AddressLineTwoInput: FC<AddressLineTwoInputProps> = ({ defaultValue }) => (
+    <TextInput
+        className='col-span-full'
+        type='text'
+        label='Address Line 2'
+        name='address-line-2'
+        fieldMappingKey='line2'
+        defaultValue={defaultValue}
+    />
+);
+
+type CityInputProps = {
+    defaultValue?: string;
+};
+
+export const CityInput: FC<CityInputProps> = ({ defaultValue }) => (
+    <TextInput
+        className='sm:col-span-2 sm:col-start-1'
+        type='text'
+        label='City'
+        name='city'
+        fieldMappingKey='city'
+        defaultValue={defaultValue}
+    />
+);
+
+type StateInputProps = {
+    defaultValue?: string;
+};
+
+export const StateInput: FC<StateInputProps> = ({ defaultValue }) => (
+    <TextInput
+        className='sm:col-span-2'
+        type='text'
+        label='State / Province'
+        name='state'
+        fieldMappingKey='state'
+        defaultValue={defaultValue}
+    />
+);
+
+type PostalCodeInputProps = {
+    defaultValue?: string;
+};
+
+export const PostalCodeInput: FC<PostalCodeInputProps> = ({ defaultValue }) => (
+    <TextInput
+        className='sm:col-span-2'
+        type='text'
+        label='ZIP / Postal code'
+        name='postal-code'
+        fieldMappingKey='postalCode'
+        defaultValue={defaultValue}
+    />
+);
```

Comment:
And this one is also quite repetitive I think, but not quite straightforward to condense maybe as there are some more things that differ between fields. But DRYer seems definitely possible nonetheless.

Reply:
Marking this as resolved as it's a duplicate of below